addons/mail/static/src/models/thread_cache/thread_cache.js

```diff
@@ -282,27 +282,35 @@ function factory(dependencies) {
             const moderated_channel_ids = this.thread.moderation
                 ? [this.thread.id]
                 : undefined;
-            const messages = await this.async(() =>
-                this.env.models['mail.message'].performRpcMessageFetch(
-                    domain,
-                    limit,
-                    moderated_channel_ids,
-                    context,
-                )
-            );
-            this.update({
-                fetchedMessages: [['link', messages]],
-                isLoaded: true,
-                isLoading: false,
-            });
-            if (!extraDomain && messages.length < limit) {
-                this.update({ isAllHistoryLoaded: true });
+            try {
+                const messages = await this.async(() =>
+                    this.env.models['mail.message'].performRpcMessageFetch(
+                        domain,
+                        limit,
+                        moderated_channel_ids,
+                        context,
+                    )
+                );
+                this.update({
+                    fetchedMessages: [['link', messages]],
+                    isLoaded: true,
+                    isLoading: false,
+                });
+                if (!extraDomain && messages.length < limit) {
+                    this.update({ isAllHistoryLoaded: true });
+                }
+                this.env.messagingBus.trigger('o-thread-cache-loaded-messages', {
+                    fetchedMessages: messages,
+                    threadCache: this,
+                });
+                return messages;
+            } catch(e) {
+                this.update({
+                    isLoaded: true,
+                    isLoading: false,
+                });
+                return [];
```

Comment:
Also this return value does not seem correct, the caller that is then triggering `new-messages-loaded` should not trigger it if this failed.